hapi/hedera-protobufs/services/state/blockstream/block_stream_info.proto

```diff
@@ -48,36 +53,36 @@ option java_multiple_files = true;
 message BlockStreamInfo {
     /**
      * A block number.<br/>
-     * This is the block number of the last completed block.
+     * This is the current block number.
      */
-    uint64 last_block_number = 1;
+    uint64 block_number = 1;
 
     /**
-     * A block hash value.<br/>
-     * This is the hash of the last completed block.
-     */
-    bytes last_block_hash = 2;
-
-    /**
-     * A consensus time for the last completed block.
-     * This is used to determine if this block was the first across an
+     * A consensus time for the current block.<br/>
+     * This is the _first_ consensus time in the current block, and
+     * is used to determine if this block was the first across an
      * important boundary in consensus time, such as UTC midnight.
      * This may also be used to purge entities expiring between the last
      * block time and this time.
      */
-    proto.Timestamp last_block_time = 3;
+    proto.Timestamp block_time = 2;
 
     /**
      * A concatenation of hash values.<br/>
      * This combines several trailing output block item hashes and
-     * is used as a seed value for a pseudo-random number generator.
+     * is used as a seed value for a pseudo-random number generator.<br/>
+     * This is also requiried to implement the EVM `PREVRANDAO` opcode.
      */
-    bytes trailing_output_hashes = 4;
+    bytes trailing_output_hashes = 3;
 
     /**
      * A concatenation of hash values.<br/>
-     * This combines the last 256 trailing block hashes, and is required to
-     * support the EVM `BLOCKHASH` opcode.
+     * This field combines up to 256 trailing block hashes.
+     * <p>
+     * If this message is for block number N, then the earliest available
+     * hash SHALL be for block number N-256.<br/>
+     * The latest available hash SHALL be for block N-1.<br/>
+     * This is REQUIRED to implement the EVM `BLOCKHASH` opcode.
```

Comment:
Same as above: it would be nice to know the min and max allowed/required length for this field, for the same reason.

Reply:
Not sure I fully understand this request...do you have suggested wording?

Our block hashes are SHA-384 hashes, though we only use the last 32 bytes to match the length of an Ethereum block hash.

Reply:
It's unclear from the specification of the field what is the minimum size of the `bytes` field, and what is the maximum size of the field. For example, similar to a field just above this one, if it has to be at least 256 bits (aka 32 bytes) long, then it's worth documenting it, so that consumers of this field can rely on the data in the field.

Reply:
Perhaps something like this?
```suggestion
     * This field combines up to 256 trailing block hashes.
     * <p>
     * If this message is for block number N, then the earliest available
     * hash SHALL be for block number N-256.<br/>
     * The latest available hash SHALL be for block N-1.<br/>
     * This is REQUIRED to implement the EVM `BLOCKHASH` opcode.
     * <p>
     * <h5>Field Length</h5>
     * Each hash value SHALL be the trailing 265 bits of a SHA2-384 hash.<br/>
     * The length of this field SHALL be an integer multiple of 32 bytes.<br/>
     * This field SHALL be at least 32 bytes.<br/>
     * The maximum length of this field SHALL be 8192 bytes.
```